### PR TITLE
APC minor tweaks

### PR DESCRIPTION
**Pull request by Mysak0CZ**
# This is my first PR, so please carefully watch for any mistakes I made, and notify my, thank you.
- I moved some things in APC's construction code.
- Emaged and malf-AI-hacked APCs can now be fixed replacing theyir board (old one destroyed in process).  - We emag and hack APC's board, not frame
- You can now add/remove APC's wires even if board is secured
- Removing APC's and SMES's terminal no longer ignore current tool's speed
- You can repair APC's cover (only if APC is not compleatly broken) by using APC frame on it, while electronicks are still in place
- You can no longer unlock AI-hacked APCs, throught AI still can
- closing APC's cover will lock it

:cl:
fix: You no longer need to deconstruct emagged (or AI-hacked) APC to fix them, replacing board is sufficient
fix: You can no longer unlock AI-hacked APCs
fix: Removing APC's and SMES's terminal no longer ignore current tool's speed
add: You can repair APC's cover (only if APC is not compleatly broken) by using APC frame on it
add: closing APC's cover will lock it
/:cl:

I already compleatly tested my code on my local server

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
Made no sense for an electronic being to hack the metal of an APC, good change :+1:

**Comment by GunHog** [comment]
Logical change, good stuff.

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -380,37 +380,60 @@
 
 	if(issilicon(user) && get_dist(src,user)>1)
 		return src.attack_hand(user)
-	if (istype(W, /obj/item/weapon/crowbar) && opened)
-		if (has_electronics==1)
-			if (terminal)
-				user << "<span class='warning'>Disconnect the wires first!</span>"
+	if (istype(W, /obj/item/weapon/crowbar)) //Using crowbar
+		if (opened) // a) on open apc
+			if (has_electronics==1)
```
Comment (optimumtact) [inline code/modules/power/apc.dm]: If this is only ever 0 or 1 just do if(has_electronics)

File: code/modules/power/apc.dm [hunk 2]
```diff
@@ -380,37 +380,60 @@
...
+				if (terminal)
+					user << "<span class='warning'>Disconnect the wires first!</span>"
+					return
+				playsound(src.loc, W.usesound, 50, 1)
+				user << "<span class='notice'>You are trying to remove the power control board...</span>" //lpeters - fixed grammar issues
+				if(do_after(user, 50/W.toolspeed, target = src))
+					if (has_electronics==1)
+						has_electronics = 0
```
Comment (optimumtact) [inline code/modules/power/apc.dm]: Again if this is only a boolean value then change it to use TRUE/FALSE defines

Comment (optimumtact) [inline code/modules/power/apc.dm]: same goes for a few other places in this code if you're up to it.

File: code/modules/power/apc.dm [hunk 3]
```diff
@@ -535,16 +571,19 @@
...
-		user.visible_message(\
-			"[user.name] has replaced the damaged APC frontal panel with a new one.",\
-			"<span class='notice'>You replace the damaged APC frontal panel with a new one.</span>")
-		qdel(W)
-		update_icon()
-	else if (istype(W, /obj/item/wallframe/apc) && opened && ((stat & BROKEN) || malfhack))
+
+	else if (istype(W, /obj/item/wallframe/apc) && opened)
+		if (!(stat & BROKEN || opened==2 || obj_integrity < max_integrity)) // There is nothing to repair
+			user << "<span class='warning'>You found no reason for repairing this APC</span>"
```
Comment (optimumtact) [inline code/modules/power/apc.dm]: why not return here, then you can skip the stat & BROKEN check down below?

**Review by optimumtact — COMMENTED** [review]
Some minor things but it otherwise looks fine.

File: code/modules/power/apc.dm (hunk 1 above)
Comment (Mysak0CZ) [inline code/modules/power/apc.dm]: no, there is 0=none;1=inserted;2=secured, we want to remove usecured electronicks with crowbar, but with secured we want to close the panel

File: code/modules/power/apc.dm (hunk 3 above)
Comment (Mysak0CZ) [inline code/modules/power/apc.dm]: Whoops, yes there shoud be return, fixing it...

**Comment by Mysak0CZ** [comment]
Ok, i tested it, from my side it looks good

File: code/modules/power/apc.dm (hunk 1 above)
Comment (optimumtact) [inline code/modules/power/apc.dm]: Okay cool

**Comment by optimumtact** [comment]
Looks good to me, thanks for your work

**Comment by HankMoodyExperience** [comment]
yeah apc destroyed mission accomplished

File: code/modules/power/apc.dm (hunk 1 above)
Comment (nullbear) [inline code/modules/power/apc.dm]: "should use defines" lol.

**Pull request merged**